swift/Picture.swift

```diff
@@ -0,0 +1,130 @@
+        guard tvg_picture_load(pointer, path) == TVG_RESULT_SUCCESS else {
+            throw ThorVGError.failedToLoadFromPath
+        }
+    }
+
+    /// Loads a picture from a given string, typically containing SVG or Lottie data.
+    ///
+    /// Use the `mimeType` to indicate the data format to ThorVG for correct parsing and rendering.
+    func load(fromString string: String, mimeType: String) throws {
+        guard let cString = string.cString(using: .utf8),
+              tvg_picture_load_data(pointer, cString, UInt32(cString.count), mimeType, false) == TVG_RESULT_SUCCESS
+        else {
+            throw ThorVGError.failedToLoadFromString
+        }
+    }
+
+    /// Retrieves the size of the picture.
+    func getSize() -> CGSize {
+        var width: Float = 0
+        var height: Float = 0
+        tvg_picture_get_size(pointer, &width, &height)
+        return CGSize(width: Double(width), height: Double(height))
+    }
+
+    /// Sets the size of the picture.
+    func setSize(_ size: CGSize) {
+        tvg_picture_set_size(pointer, Float(size.width), Float(size.height))
+    }
+
+    /// Retrieves the current transform applied to the picture.
+    func getTransform() -> CGAffineTransform {
+        var matrix = Tvg_Matrix()
+        tvg_paint_get_transform(pointer, &matrix)
+
+        return CGAffineTransform(
+            CGFloat(matrix.e11), CGFloat(matrix.e21), CGFloat(matrix.e12),
+            CGFloat(matrix.e22), CGFloat(matrix.e13), CGFloat(matrix.e23)
+        )
+    }
+
+    /// Applies a new transform to the picture.
+    func setTransform(_ transform: CGAffineTransform) {
+        var matrix = Tvg_Matrix(
+            e11: Float(transform.a), e12: Float(transform.c), e13: Float(transform.tx),
+            e21: Float(transform.b), e22: Float(transform.d), e23: Float(transform.ty),
+            e31: 0, e32: 0, e33: 1
+        )
+
+        tvg_paint_set_transform(pointer, &matrix)
+    }
+
+    /// Rotates the picture about its center point.
+    ///
+    /// Note: We need to set the transform manually here because ThorVG's
+    /// default transformation behaviour rotates about the origin (not the center)
+    /// which is by default, the top-left of the picture.
+    func rotateAboutCenter(_ degrees: Double) {
+        let size = getSize()
+        let cx = size.width / 2
+        let cy = size.height / 2
+
+        let radians = degrees * .pi / 180.0
+
+        var transform = getTransform()
+        transform = transform
+            .translatedBy(x: cx, y: cy)
+            .rotated(by: radians)
+            .translatedBy(x: -cx, y: -cy)
+        setTransform(transform)
+    }
+
+    /// Crops the picture to a specified rectangle, maintaining its perceived size.
+    ///
+    /// Note: We "crop" the picture by scaling it relative to it's original size.
+    /// This means that we maintain the same perceived size of the picture before and after cropping.
+    func crop(_ crop: CGRect) {
```

Comment:
I'm not sure I understand what this function does. More specifically, what do you mean by "maintaining its perceived size"?

Reply:
Have a look at the images in [this PR](https://github.com/andyf-canva/thorvg-swift/pull/4). 

The issue is: once we crop the image, we're then left with a of dead space in the Canvas. So what this cropping behaviour does is once we've cropped the content, scale it back up to the size of the Canvas (or the original `Picture` size).